docs/sources/flow/reference/components/pyroscope.ebpf.md

```diff
@@ -0,0 +1,178 @@
+---
+title: phlare.ebpf
+labels:
+  stage: beta
+---
+
+# phlare.ebpf
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" >}}
+
+`pyroscope.ebpf` configures an ebpf profiling job for the current host. The collected performance profiles are forwarded
+to
+the list of receivers passed in
+`forward_to`.
+
+Multiple `pyroscope.ebpf` components can be specified by giving them different labels, however it is not recommended.
+
+## Usage
+
+```river
+pyroscope.ebpf "LABEL" {
+  forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+The component configures and starts a new ebpf profiling job to collect performance profiles from the current host.
+
+The following arguments can be used to configure a `pyroscope.ebpf`. Only the
+`forward_to` field is required and any omitted fields take their default
+values.
+
+| Name                      | Type                     | Description                                                            | Default | Required |
+|---------------------------|--------------------------|------------------------------------------------------------------------|---------|----------|
+| `forward_to`              | `list(ProfilesReceiver)` | List of receivers to send collected profiles to.                       |         | yes      |  
+| `targets`                 | `list(map(string))`      | List of targets to group profiles by container id                      |         | no       |   
```

Comment:
If this only works with containers we should specify it in the documentation summary.

Reply:
If this is the case I think we should also create an issue to support non-container workload.

We could probably create a service discovery finds process, allow to filter them and send target with the associated `___process_pid__`.

Reply:
I think it also works without containers for the whole system if you set `targets_only=false` and `default_target={service_name="System"}` but this took me some time to click.

Reply:
yeah I want to make it more explicit via a new service discovery component.